pybamm/solvers/base_solver.py

```diff
@@ -781,6 +784,13 @@ def solve(
         # Set up (if not done already)
         timer = pybamm.Timer()
         if model not in self.models_set_up:
+            if len(self.models_set_up) > 0:
```

Comment:
I think `self.models_set_up` (a dict) could now be `self._model_set_up` (a model) since only one model is allowed